pilot/pkg/networking/core/v1alpha3/networkfilter.go

```diff
@@ -166,6 +166,8 @@ func buildOutboundNetworkFiltersWithWeightedClusters(node *model.Proxy, routes [
 	// TODO: Need to handle multiple cluster names for Redis
 	clusterName := clusterSpecifier.WeightedClusters.Clusters[0].Name
 	tcpFilter := setAccessLogAndBuildTCPFilter(push, node, proxyConfig)
+
+	push.GatewayClusterNames[clusterName] = true
```

Comment:
It doesn't seem appropriate to modify push context after creation. AFAIK currently its unchanged after initialization

Reply:
updated!